src/main/kotlin/no/nav/syfo/client/ereg/EregClient.kt

```diff
@@ -0,0 +1,65 @@
+package no.nav.syfo.client.ereg
+
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.httpClientDefault
+import no.nav.syfo.util.bearerHeader
+import org.slf4j.LoggerFactory
+
+class EregClient(
+    private val azureAdClient: AzureAdClient,
+    private val isproxyClientId: String,
+    baseUrl: String,
+) {
+    private val httpClient = httpClientDefault()
+
+    private val eregOrganisasjonUrl: String = "$baseUrl$EREG_PATH"
+
+    suspend fun organisasjon(
+        callId: String,
+        orgNr: String,
+    ): EregOrganisasjonResponse? {
+        val systemToken = azureAdClient.getSystemToken(
+            scopeClientId = isproxyClientId,
+        )?.accessToken
+            ?: throw RuntimeException("Failed to request Organisasjon from Isproxy-Ereg: Failed to get system token from AzureAD")
+
+        try {
+            val url = "$eregOrganisasjonUrl/$orgNr"
+            val response: EregOrganisasjonResponse = httpClient.get(url) {
+                header(HttpHeaders.Authorization, bearerHeader(systemToken))
+                accept(ContentType.Application.Json)
+            }
+            COUNT_CALL_EREG_ORGANISASJON_SUCCESS.increment()
+            return response
+        } catch (e: ClientRequestException) {
+            handleUnexpectedResponseException(e.response, e.message, callId)
+        } catch (e: ServerResponseException) {
+            handleUnexpectedResponseException(e.response, e.message, callId)
+        }
+        return null
+    }
+
+    private fun handleUnexpectedResponseException(
+        response: HttpResponse,
+        message: String?,
+        callId: String,
+    ) {
+        log.error(
+            "Error while requesting Response from Ereg {}, {}, {}",
+            StructuredArguments.keyValue("statusCode", response.status.value.toString()),
+            StructuredArguments.keyValue("message", message),
+            StructuredArguments.keyValue("callId", callId),
+        )
+        COUNT_CALL_EREG_ORGANISASJON_FAIL.increment()
+    }
+
+    companion object {
+        const val EREG_PATH = "/api/v1/ereg/ereg/api/v1/organisasjon"
```

Comment:
Skal ikke dette bare være: /api/v1/ereg ?

Reply:
Det er det som er url i Isproxy nå. Der har jeg fulgt mønsteret som er brukt for dokdist:
"api/v1/<tjeneste>/<pathTilEndepunktITjenste>"

<pathTilEndepunktITjenste> er altså "/ereg/api/v1/organisasjon""

Reply:
OK, klarer ikke helt å se at det er dette vi har gjort for dokdist. Men det kan godt være at dette er en god ide likevel 👍

Reply:
Sant, det er ikke helt det samme. Sa litt feil der. Men blir lettere om endepunktet man kaller i isproxy er ca. det samme som isproxy kaller ut. Men jeg tenker det blir lettere å bytte over. Altså at man kun må fjerne et postfix på urlen og så er alt annet likt.

Kan evt. fjerne ereg foran så man slipper forvirring, men postfix/pathIEkteTjeneste tenker jeg i hvert fall er lurt.

Reply:
Url for api i isproxy til dokdist er vel `/api/v1/dokdist/distribuerjournalpost` ? Så jeg tenker også det bør være `/api/v1/ereg/organisasjon` her 🤷‍♂️

Reply:
Vi må tilpasse i alle fall når vi skal gå rett på riktig endepunkt og ikke via isproxy. I mellomtiden synes jeg /api/v1/ereg/organisasjon er minst forvirrende både her og i isproxy.

Reply:
Da har jeg oppdatert path til isproxy til: `const val EREG_PATH = "/api/v1/ereg/organisasjon"`